test/test_datasets.py

```diff
@@ -595,6 +595,18 @@ def test_attr_names(self):
         with self.create_dataset() as (dataset, info):
             assert tuple(dataset.attr_names) == info["attr_names"]
 
+    def test_images_names_split(self):
+        splits = ["train", "valid", "test"]
+        with self.create_dataset(split='all') as (dataset, _):
+            all_imgs_names = dataset.filename
+
+        merged_imgs_names = []
+        for split in splits:
+            with self.create_dataset(split=split) as (dataset, _):
+                merged_imgs_names += dataset.filename
```

Comment:
1. This relies on the fact that the images of the splits are back to back, i.e. the ordering is important. Since we actually don't want to test this here, we might as well use a set to ignore the ordering.
2. Nitpick: since `splits` is only used once, I would prefer to inline it or at least move it closer to this loop


```suggestion
        merged_imgs_names = set()
        for split in ["train", "valid", "test"]:
            with self.create_dataset(split=split) as (dataset, _):
                merged_imgs_names.update(dataset.filename)
```

Reply:
Agree as to it is the better way for its purposes.